Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -51,19 +51,19 @@ async function run() {
         tl._writeLine(tl.loc('GotconnectiondetailsforazureRMWebApp0', webAppName));
         
         if(xmlTransformationAndVariableSubstitution && xmlTransformation) {
-            if(canUseWebDeploy(useWebDeploy)) {
+            if(tl.osType().match(/^Win/)) {
```

Comment:
If this option is enabled for a wrong platform you should be giving appropriate guidance / error message.